metric-anomaly-data-model/src/main/proto/org/hypertrace/alert/engine/eventcondition/config/service/v1/eval_rule.proto

```diff
@@ -2,17 +2,23 @@ syntax = "proto3";
 
 option java_multiple_files = true;
 
-package org.hypertrace.alert.engine.metric.anomlay.rule.config.service.v1;
+package org.hypertrace.alert.engine.eventcondition.config.service.v1;
 
-import "org/hypertrace/metric/anomaly/rule/config/service/v1/alert_rule_enum.proto";
+import "org/hypertrace/alert/engine/eventcondition/config/service/v1/eval_rule_enums.proto";
 
-message MetricAnomalyAlertRule {
+
+message MetricAnomalyEventCondition {
   string name = 1;
   repeated Label labels = 2;
+  MetricAnomalyEvalRule eval_rule = 3;
+  NotificationCondition notification_condition = 4;
```

Comment:
What kind of stuff would go in here? The baseline conditions already have severity, I would think the notification really pairs with that. A rule builder would look like - a metric selection, then one or more clauses like "when value crosses this threshold, do X"

Reply:
Here, I was thinking it will represent a discrete event where it will help to indicate the change in alert condition like critical to warning or normal to warning, etc. It will also help in a situation where the alert state continues for a longer period (e.g latency was high for 1 hrs). In such cases,  the evaluation engine will continue to report anomalies for the next 1 hrs as  - critical. And, so this can be utilized for silencing by the notifier.

Reply:
So the change in condition makes sense as a use case, and I could see it modeled in one of two ways. In one way (I think this is what you were thinking), you basically have a set of rules generating events with metadata (in other words, discretizing them), and then a separate rule on how to alert on those, the latter being similar to our event rules. What I think needs a bit more thought there (and this is a bit what I was getting at with the original comment) is that the general scheme sounds a lot like a state machine where the event conditions are the states and we're putting notifications on state transitions. We could consider if it would be more clear to a user to instead use the various thresholds as our states, rather than the indirection of severity in between. In other words, is it clearer for a user to say "When metric X crosses threshold Y: then high severity" and "When high severity, then send email" vs "When metric X crosses threshold Y: then send email". I do think the former can model more, but am concerned it may be too confusing. We of course could use the more complex modeling and simplify the UX though.


The notification throttle (although it too makes sense as a use case), is already part of the parent rule and need not be modeled as part of a condition object - the throttling of notification should be handled at the generic, condition-agnostic layer.

Reply:
As we want to express,

- alert me when metric X crosses threshold Y as a warning
- alert me when metric X crosses threshold Z as a critical

so as you describe, let's go with multiple thresholds as states and the evaluation engine will generate an event on each state transition. Initially, here severity will be more like a label for that threshold. So, I moved severity along with StaticThreshold.